### PR TITLE
Add exporting Summary metric to Prometheus

### PR DESCRIPTION
**Pull request by cijothomas**
Added capability to export Summary to Prometheus.

(The Summary here refers to OT summary of MinMaxSumCount. Prometheus Summary is capable of storing percentiles as well. OT dont have that yet.)

### CODE REVIEW

File: test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs [hunk 1]
```diff
@@ -76,17 +57,17 @@ public async Task E2ETestMetricsHttpServerAsync()
...
                 Assert.Equal(HttpStatusCode.OK, response.StatusCode);
                 var responseText = response.Content.ReadAsStringAsync().Result;
-                Assert.Contains("testCounter{dim1=\"value1\"}", responseText);
-                Assert.Contains("testCounter{dim1=\"value2\"}", responseText);
+                this.output.WriteLine($"Respone from metrics API is \n {responseText}");
+                ValidateResponse(responseText);
                 metricsHttpServer.Stop();
             }
-        }
+        }
```
Comment (MikeGoldsmith) [inline test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs:70]: nit: whitespace

File: test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs [hunk 2]
```diff
@@ -119,12 +100,13 @@ public async Task E2ETestMiddleware()
...
+                Task.Delay(waitDuration).Wait();
                 response = await client.GetAsync("/metrics");
                 Assert.Equal(HttpStatusCode.OK, response.StatusCode);
-                var s = response.Content.ReadAsStringAsync().Result;
-                Assert.Contains("testCounter{dim1=\"value1\"}", response.Content.ReadAsStringAsync().Result);
-                Assert.Contains("testCounter{dim1=\"value2\"}", response.Content.ReadAsStringAsync().Result);
+                var responseText = response.Content.ReadAsStringAsync().Result;
+                this.output.WriteLine($"Respone from metrics API is \n {responseText}");
+                ValidateResponse(responseText);
+
```
Comment (MikeGoldsmith) [inline test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs:109]: nit: whitespace

File: test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs [hunk 3]
```diff
@@ -76,17 +57,17 @@ public async Task E2ETestMetricsHttpServerAsync()
             }
             finally
             {                
-                Task.Delay(400).Wait();
+                Task.Delay(waitDuration).Wait();
```
Comment (MikeGoldsmith) [inline test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs:60]:
```suggestion
                await Task.Delay(waitDuration);
```

File: test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs [hunk 4]
```diff
@@ -119,12 +100,13 @@ public async Task E2ETestMiddleware()
                 var response = await client.GetAsync("/foo");
                 Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
 
-                Task.Delay(2000).Wait();
+                Task.Delay(waitDuration).Wait();
```
Comment (MikeGoldsmith) [inline test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs:103]:
```suggestion
                await Task.Delay(waitDuration);
```

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs [hunk 5]
```diff
@@ -66,7 +62,50 @@ public static void WriteMetricsCollection(this PrometheusExporter exporter, Stre
...
+                                metricValueBuilder = metricValueBuilder.WithValue(longValueSum);
+                                metricValueBuilder.WithLabel(label.Key, label.Value);
+
+                                metricValueBuilder = builder.AddValue();
+                                metricValueBuilder.WithName(metric.MetricName + "_count");
+                                metricValueBuilder = metricValueBuilder.WithValue(longValueCount);
+                                metricValueBuilder.WithLabel(label.Key, label.Value);
+
+                                metricValueBuilder = builder.AddValue();
+                                metricValueBuilder.WithName(metric.MetricName);
```
Comment (MikeGoldsmith) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:96]: Should min and max metric names also have a suffix like `_sum` & `_count`?

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs [hunk 6]
```diff
@@ -100,7 +139,50 @@ public static void WriteMetricsCollection(this PrometheusExporter exporter, Stre
...
```
Comment (MikeGoldsmith) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:173]: Should min and max metric names also have a suffix like _sum & _count?

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs [hunk 7]
```diff
@@ -66,7 +62,50 @@ public static void WriteMetricsCollection(this PrometheusExporter exporter, Stre
...
+                                In future, when OT implements more aggregation algorithms,
+                                this section will need to be revisited.
+                                Sample output:
+                                MyMeasure_sum{dim1="value1"} 750 1587013352982
+                                MyMeasure_count{dim1="value1"} 5 1587013352982
+                                MyMeasure{dim1="value2",quantile="0"} 150 1587013352982
+                                MyMeasure{dim1="value2",quantile="1"} 150 1587013352982
+                                */
+                                var metricValueBuilder = builder.AddValue();
+                                metricValueBuilder.WithName(metric.MetricName + "_sum");
```
Comment (MikeGoldsmith) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:86]: Please move reused strings into private const: eg _sum, _count, quantile, 0, 1

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs (hunk 5 above)
Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:96]: No, min and max are exported as quantiles 0 and 1 respectively.

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs (hunk 6 above)
Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:173]: No, min and max are exported as quantiles 0 and 1 respectively.

**Comment by cijothomas** [comment]
@MikeGoldsmith Comments addressed. Please review and merge! thanks

**Pull request merged**